src/dds/rs-dds-device-proxy.cpp

```diff
@@ -160,11 +151,14 @@ dds_device_proxy::dds_device_proxy( std::shared_ptr< context > ctx, std::shared_
                 _software_sensors.push_back( sensor_info.proxy );
             }
             auto stream_type = to_rs2_stream_type( stream->type_string() );
-            sid_index sidx( stream_type, counts[stream_type]++ );
+            int index = get_index_from_stream_name( stream->name() );
+            sid_index sidx( environment::get_instance().generate_stream_id(), index);
+            sid_index type_and_index( stream_type, index );
             _stream_name_to_librs_stream[stream->name()]
                 = std::make_shared< librealsense::stream >( stream_type, sidx.index );
             sensor_info.proxy->add_dds_stream( sidx, stream );
             _stream_name_to_owning_sensor[stream->name()] = sensor_info.proxy;
+            type_and_index_to_dds_stream_sidx.insert( { type_and_index, sidx }  );
```

Comment:
Better to use `emplace`?

Reply:
`emplace` does not accept initializer list as a parameter, I will have to use `make_pair`. I don't know what is more efficient (but negligible anyway), but I think that like this it is more readable.